Shogi2 - přestavba struktury kódu

danielz9999 opened the pull request:
Ve větvi shogi2 jde o pokus začít cel kód znovu, s cílem zlepšit objektovou hiearchii, zavést soukromé atributy, správné využití get/set metod etc.. V této verzi zatím žádný pohyb, pouze šachovnice a na ní jedna figurka.

danielz9999 committed:
Shogi2 - přestavba struktury kódu

Ve větvi shogi2 jde o pokus začít cel kód znovu, s cílem zlepšit objektovou hiearchii, zavést soukromé atributy, správné využití get/set metod etc.. V této verzi zatím žádný pohyb, pouze šachovnice a na ní jedna figurka.

danielz9999 committed:
Delete Dvojice.java

danielz9999 committed:
Delete GameRules.java

danielz9999 committed:
Delete GoldGeneral.java

danielz9999 committed:
Delete King.java

danielz9999 committed:
Delete Knight.java

danielz9999 committed:
Delete Lancer.java

danielz9999 committed:
Delete SilverGeneral.java

danielz9999 committed:
Delete nullPiece.java

danielz9999 committed:
Delete pole.java

danielz9999 committed:
Delete testPiece.java

danielz9999 committed:
Přidání chytrého pohybu figurek

Přidáním obechných pohybových funkcí založených na předělaném shogi2 kódu, figurky (zatím pouze pěšáci) nyní nemohou pohybovat skrze přátelské figurky, což je zlepšení oproti staršímu kódu.

danielz9999 committed:
Přidány všechny hrací figurky, úprava pohybu figurek

Byly doplněny všechny zbývající tříd figurek a jejich odpovídající getMoves metoda, několik úprav normální pohybové funkce, která předtím nefungovala správně, zároveň bylo spraveno odečítání pozic z listu pozic, což zabraňovalo pohybům na dříve obsazená políčka.

lillieMaiBauer commented on piece.java:
```diff
@@ -1,30 +1,54 @@
-//obecná parent třída všech figurek
+import java.util.ArrayList;
 
 public class piece {
```
I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [Nm: Class names should start with an upper case letter (NM_CLASS_NAMING_CONVENTION)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#nm-class-names-should-start-with-an-upper-case-letter-nm-class-naming-convention).
Class names should be nouns, in mixed case with the first letter of each internal word capitalized. Try to keep your class names simple and descriptive. Use whole words-avoid acronyms and abbreviations (unless the abbreviation is much more widely used than the long form, such as URL or HTML).

danielz9999 committed:
Přidáno vylepšování figurek

Když se figurky dostanou na vylepšovací políčka (poslední 3 řady na druhém konci šachovnice), vyskočí okénko s možnostmi Y/N. Když hráč zmáčkne YES, tak se na místě dané figurky vygeneruje nová vylepšená, pokud je zmáčknuto NO, hra pokračuje. Zároveň je zde automatické vylepšování figurek, které už se nmůžou dále hýbat.
